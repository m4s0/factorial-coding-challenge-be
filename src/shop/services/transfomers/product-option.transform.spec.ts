import { ProductOption } from '@Shop/entities/product-option.entity';
import { transformProductOption } from './product-option.transform';

describe('transformProductOption', () => {
  it('should transform product option correctly', () => {
    const productOption = {
      id: 'fb5d0e88-8014-4883-82d5-94d0af8e723f',
      name: 'Size',
      displayName: 'Product Size',
      basePrice: 10.99,
      isActive: true,
      createdAt: new Date('2024-01-01'),
      updatedAt: new Date('2024-01-02'),
    } as ProductOption;

    const result = transformProductOption(productOption);

    expect(result).toMatchSnapshot();
  });
});
