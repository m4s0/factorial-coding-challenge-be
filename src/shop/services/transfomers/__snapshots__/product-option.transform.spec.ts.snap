// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformProductOption should transform product option correctly 1`] = `
{
  "basePrice": 10.99,
  "createdAt": "2024-01-01T00:00:00.000Z",
  "displayName": "Product Size",
  "id": "fb5d0e88-8014-4883-82d5-94d0af8e723f",
  "isActive": true,
  "name": "Size",
  "updatedAt": "2024-01-02T00:00:00.000Z",
}
`;
